test/e2e/suite/approval/userinfo.go

```diff
@@ -73,7 +76,7 @@ var _ = framework.CertManagerDescribe("UserInfo CertificateRequests", func() {
 		By("Should error when attempting to update UserInfo fields")
 		cr.Spec.Username = "abc"
 		cr.Spec.UID = "123"
-		cr, err = f.CertManagerClientSet.CertmanagerV1().CertificateRequests(f.Namespace.Name).Update(context.TODO(), cr, metav1.UpdateOptions{})
+		_, err = internalcertificaterequests.Apply(context.Background(), f.CertManagerClientSet, fieldManager, cr)
```

Comment:
What's the difference between `context.TODO` and `context.Background` ? Isn't there a cancellable context we should use in these tests? If you agree, then maybe add a TODO item to fix that.

Reply:
Can you think of some cases where we would need a cancellable context in tests? I've been thinking about that too, but then it seems like in all cases where there would be a need to time out the connection, the test would time out and clean up resources anyway?
As per my understanding there is no difference between `context.TODO` and `context.Background` functionally, but the first signals that it is not yet decided what context should be used https://pkg.go.dev/context#TODO

Reply:
Thanks for linking that, I've been reading that and the linked discussions.
At the moment I still don't have a clear picture of what resources (kube resources, goroutines, open connections to apiserver etc) we could potentially have still hanging around in what cases (i.e a spec failure) that we could clean up sooner if there was a cancellable context. Will keep looking at this.
Specifically in case of those apply calls, there will be [the default 34s timeout](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go#L56) I guess.

I also think we should now upgrade to Ginkgo to v2 so that we can benefit from any improvements in that version https://github.com/cert-manager/cert-manager/issues/4940